com.ibm.streamsx.topology/opt/python/packages/streamsx/topology/topology.py

```diff
@@ -49,9 +49,14 @@ def sink(self, func):
 
     def filter(self, func):
         """
-        Filters tuples from a stream using the supplied function.
-        For each tuple on the stream the function is called passing
-        the tuple, if the function return evalulates to true the
+        Filters tuples from a stream using the supplied callable 'func'.
+        The callable input must be either 
+        1) the name of a top-level function defined using 'def' that takes a single parameter for the tuple, or
+        2) an instance of a top-level callable class that implements the method '__call__(self, tuple)'.  Using a
```

Comment:
What's a "top-level" function or class?

I don't see a standard definition in my python book.

I found this: https://docs.python.org/3.5/reference/toplevel_components.html
but it never defines what top-level means.